python/python-psi-impl/src/com/jetbrains/python/lexer/PyFStringLiteralLexer.kt

```diff
@@ -10,23 +11,24 @@ class PyFStringLiteralLexer(fStringTextToken: IElementType) : PyStringLiteralLex
     assert(PyTokenTypes.FSTRING_TEXT_TOKENS.contains(fStringTextToken))
   }
 
-  override fun locateToken(start: Int): Int {
-    if (start >= myBufferEnd) {
-      return myBufferEnd
-    }
-    
-    if (myBuffer[start] == '\\') {
-      return locateEscapeSequence(start)
-    }
-    else {
-      val nextBackslashOffset = CharArrayUtil.indexOf(myBuffer, "\\", start + 1, myBufferEnd)
-      return if (nextBackslashOffset >= 0) nextBackslashOffset else myBufferEnd
-    }
+  override fun locateToken(start: Int): Int = when {
+    start >= myBufferEnd -> myBufferEnd
+    myBuffer[start] == '\\' -> locateEscapeSequence(start)
+    isDoubleCurley() -> start + 2
+    else -> myBuffer.subSequence(start+1..myBufferEnd).findAnyOf(setOf("\\", "{{", "}}"))?.first ?: myBufferEnd
```

Comment:
In the editor `myBuffer` is an instance of  `ImmutableText` and creating a subsequence of it is relatively cheap, it's just an immutable view at the same data. However, it's not always the case and I wouldn't depend on that. For instance, inside language injections, the buffer is just a `String` and its subsequence is actually an independent copy of the range.
In case of lexers that should be very performant, it's always safer to use a plain old indexed loop so as not to hit unexpected side-effects.